add auth_password_settings module

dhp-denero opened the pull request.

lmignon committed:
[ADD] This module initialize the session by looking for the field HTTP_REMOTE_USER in the HEADER of the HTTP request and trying^Co bind the given value to a user

lmignon committed:
port auth_from_http_remote_user to 8.0

lmignon committed:
[PEP8] line lenght is now ridiculous (80 chars)

lmignon committed:
[FIX] if the default login page is disabled, redirect with *401*

lmignon committed:
Improve module description

lmignon committed:
Addons configuration parameter is now a field part of the Authentication group from base_setup[C

lmignon committed:
Improve doc

lmignon committed:
Code simplification: remove overkill method

lmignon committed:
Improve doc

lmignon committed:
[FIX] properlyt return the Unauthorized() response to avoid warning in the log file

lmignon committed:
[FIX] always check the uid in the session and the one in the request to avoid session mismatch...

lmignon committed:
[PEP8]

lmignon committed:
[FIX] return 'None' if no user name is found in the request hearder

lmignon committed:
If HTTP_REMOTE_USER is in the request headers and no corresponding user is found in odoo always issues Unauthorized (avoid redirect to the login page)
If the uid in the session is not the same as the one from the binded HTTP_REMOTE_USER, always logout to clean up the session

sbidoul committed:
auth_http_remote_user: slight reorganization and some comments (logic remains identical)

sbidoul committed:
auth_from_http_remote_user: remove unneeded browse

sbidoul committed:
auth_http_remote_user: test if already authenticated based on login instead of uid

Avoids a database query unless authentication is actually required.

Commit:
Move mail_environment in root folder

Commit:
Use absolute imports for openerp and new Model classes

Commit:
Use cr argument instead of cursor, propagate context

Commit:
Update fields, remove deprecated 'method' argument, change 'states'
otherwise the fields are not readonly (seems that the 'states' of the
original field is kept.

Commit:
Activate the installable flag

Commit:
Indentation of the view with 2 spaces

Commit:
Remove only the attrs attribute instead of redefining the whole field

Commit:
pep8

Commit:
crm is not a dependency for mail_environment

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

sysadminmatmoz committed:
Slovene translation added

lmignon committed:
[PEP-8]

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

anthony-muschang committed:
[CHG] auth_admin_passkey: migration to 8.0

anthony-muschang committed:
[CHG] auth_admin_passkey: move description to README.md

anthony-muschang committed:
[CHG] auth_admin_passkey: rename README.md to README.rst

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

dreispt committed:
Backport v9 shell CLI command

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

dreispt committed:
New CLI commands must go in a `cli` subdirectory

dreispt committed:
Rebase from master with PR odoo/odoo#4531

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

hbrunn committed:
[ADD] auth_dynamic_groups

hbrunn committed:
[RFR] new api/views

hbrunn committed:
[FIX] pylint

hbrunn committed:
[ADD] example

hbrunn committed:
[REN] README.rst

yvaucher committed:
Merge pull request #96 from acsone/8.0-amu-migration_auth_admin_passkey

[8.0] migration of auth_admin_passkey

pedrobaeza committed:
[IMP] auth_admin_passkey: Moved icon

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

Commit:
Fix typo in manifest

pedrobaeza committed:
Merge pull request #70 from guewen/mail_environment-migr

Migration of mail_environment

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

pedrobaeza committed:
Merge pull request #43 from acsone/auth_from_http_remote_user-8.0

[ADD] Authentication via HTTP Remote User 8.0

pedrobaeza committed:
[IMP] auth_from_http_remote_user: README.rst

sebalix committed:
[ADD] Module 'auditlog' - A substitute to the deprecated 'audittrail' module

sebalix committed:
[ADD] Module 'auditlog' - A substitute to the deprecated 'audittrail' module

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/server-tools into 8.0

sebalix committed:
[FIX] Module 'auditlog' - Fix some flake8 errors

sebalix committed:
[FIX] Module 'auditlog' - Fix some flake8 errors

sebalix committed:
[FIX] Module 'auditlog' - Put the module description in a README.rst file

sebalix committed:
[FIX] Module 'auditlog' - Put the module description in a README.rst file

sebalix committed:
[ADD] Module 'auditlog' - 'auditlog.pot' file added

sebalix committed:
[ADD] Module 'auditlog' - 'auditlog.pot' file added

sebalix committed:
[IMP] Module 'auditlog' - Split the 'create_logs()' method in several ones to make overridding easier

sebalix committed:
[IMP] Module 'auditlog' - Split the 'create_logs()' method in several ones to make overridding easier

sebalix committed:
[IMP] Module 'auditlog' - Clean up vim lines

sebalix committed:
[IMP] Module 'auditlog' - Clean up vim lines

sebalix committed:
[IMP] Module 'auditlog' - Add 'name' attributes on some filters and other minor fixes

sebalix committed:
[IMP] Module 'auditlog' - Add 'name' attributes on some filters and other minor fixes

sysadminmatmoz committed:
Translation correction

pedrobaeza committed:
Merge pull request #120 from sysadminmatmoz/8.0

Slovene translations

hbrunn committed:
[REN] preliminary migration of fetchmail_attach_from_folder

hbrunn committed:
[IMP] make manual attach wizard usable again

hbrunn committed:
[ADD] placeholders

hbrunn committed:
[FIX] rename OpenERP->Odoo

hbrunn committed:
[IMP] readme

hbrunn committed:
[FIX] completely initialize the test model

sebalix committed:
[IMP] Module 'auditlog' - Removed the 'timestamp' field from the 'auditlog.log' model (standard 'create_date' field is used instead)

sebalix committed:
[IMP] Module 'auditlog' - Removed the 'timestamp' field from the 'auditlog.log' model (standard 'create_date' field is used instead)

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (based on the OCA template)

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (based on the OCA template)

sebalix committed:
[FIX] Module 'auditlog' - 'time' module imported but unused

sebalix committed:
[FIX] Module 'auditlog' - 'time' module imported but unused

hbrunn committed:
[ADD] init hook to migrate from audittrail

hbrunn committed:
[ADD] init hook to migrate from audittrail

hbrunn committed:
[IMP] ignore rules for nonexisting objects

hbrunn committed:
[IMP] ignore rules for nonexisting objects

sebalix committed:
Merge pull request #1 from hbrunn/8.0-auditlog

migration from audittrail, small precaution

hbrunn committed:
[FIX] also migrate id sequences

hbrunn committed:
[FIX] also migrate id sequences

sebalix committed:
Merge pull request #2 from hbrunn/8.0-auditlog

[FIX] also migrate id sequences

sebalix committed:
[IMP] Module 'auditlog' - Contributors updated in README.rst + AUTHORS.txt file removed

sebalix committed:
[IMP] Module 'auditlog' - Contributors updated in README.rst + AUTHORS.txt file removed

hbrunn committed:
[ADD] change xmlids of legacy rules to point to auditlog.rule

hbrunn committed:
[ADD] change xmlids of legacy rules to point to auditlog.rule

sebalix committed:
Merge pull request #3 from hbrunn/8.0-auditlog

[ADD] change xmlids of legacy rules to point to auditlog.rule

yvaucher committed:
Merge pull request #122 from hbrunn/8.0-base_field_serialized-fix-test

[FIX] completely initialize the test model

hbrunn committed:
[IMP] use new import of _, SUPERUSER_ID

hbrunn committed:
[IMP] use new import of _, SUPERUSER_ID

hbrunn committed:
[ADD] allow overriding modules to pass additional log values

hbrunn committed:
[ADD] allow overriding modules to pass additional log values

hbrunn committed:
[IMP] cache model and field ids

hbrunn committed:
[IMP] cache model and field ids

hbrunn committed:
[IMP] don't make a temporary copy of dict keys

hbrunn committed:
[IMP] don't make a temporary copy of dict keys

hbrunn committed:
[FIX] don't reset caches in register_hook

hbrunn committed:
[FIX] don't reset caches in register_hook

hbrunn committed:
[ADD] pass old_values to create_log when deleting a record

hbrunn committed:
[ADD] pass old_values to create_log when deleting a record

hbrunn committed:
[FIX] cope with no additional log values

hbrunn committed:
[FIX] cope with no additional log values

hbrunn committed:
[ADD] tests

hbrunn committed:
[ADD] tests

sebalix committed:
Merge pull request #4 from hbrunn/8.0-auditlog

8.0 auditlog

sebalix committed:
[FIX] Module 'auditlog' - local variable 'ir_model' is assigned to but never used

sebalix committed:
[FIX] Module 'auditlog' - local variable 'ir_model' is assigned to but never used

hbrunn committed:
[IMP] use a dedicated logger

hbrunn committed:
[FIX] we want object_id not to be required

hbrunn committed:
[ADD] test matching algorithms

hbrunn committed:
[FIX] flake8

hbrunn committed:
[IMP] use a model for test that is not influenced by the mail module

hbrunn committed:
[IMP] use a model for test that is not influenced by the mail module

sebalix committed:
Merge pull request #5 from hbrunn/8.0-auditlog

[FIX] Module 'auditlog' - Use a model for test that is not influenced by the mail module

sebalix committed:
[FIX] Module 'auditlog' - Remove 'active' field from __openerp__.py

sebalix committed:
[FIX] Module 'auditlog' - Remove 'active' field from __openerp__.py

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (inform that the audittrail data - if any - are migrated during the installation)

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (inform that the audittrail data - if any - are migrated during the installation)

sebalix committed:
[IMP] Module 'auditlog' - Useless lines of code removed

sebalix committed:
[IMP] Module 'auditlog' - Useless lines of code removed

pedrobaeza committed:
Merge pull request #121 from hbrunn/8.0-fetchmail_attach_from_folder

[ADD] port fetchmail_attach_from_folder

sebalix committed:
[IMP] Module 'auditlog' - Autoremove 'auditlog.log.line' records when a field is deleted (e.g. migration)

sebalix committed:
[IMP] Module 'auditlog' - Autoremove 'auditlog.log.line' records when a field is deleted (e.g. migration)

sebalix committed:
[IMP] Module 'auditlog' - Model 'auditlog.log.line', 'field_name' and 'field_description' fields changed into related fields

sebalix committed:
[IMP] Module 'auditlog' - Model 'auditlog.log.line', 'field_name' and 'field_description' fields changed into related fields

sebalix committed:
[IMP] Module 'auditlog' - Added a second test using cached data generated during the first test

sebalix committed:
[IMP] Module 'auditlog' - Added a second test using cached data generated during the first test

sebalix committed:
[FIX] Module 'auditlog' - Data dictionaries of fields and models are put in the cache instead of recordsets (a recordset contains a reference on the database cursor which is closed after each RPC query)

sebalix committed:
[FIX] Module 'auditlog' - Data dictionaries of fields and models are put in the cache instead of recordsets (a recordset contains a reference on the database cursor which is closed after each RPC query)

lepistone committed:
capture stderr as well as stdout when diagnosing

This fixes OCA/server-tools#125 because in some Ubuntu systems
lsb_release outputs to stderr as well as stdout. This message will be
printed to console, not logged nor shown among server infomation. That
way the message is lost and only pollutes unit tests output.

yvaucher committed:
Add license key in __openerp__.py

pedrobaeza committed:
Merge pull request #128 from yvaucher/8.0-add-license-key

Add license key in __openerp__.py

nbessi committed:
Merge pull request #126 from lepistone/fix-lsb-error-to-console

capture stderr as well as stdout when diagnosing

sebalix committed:
[IMP] Module 'auditlog' - Add a test regarding a resource deleted through a (2, ID) tuple on a write operation

sebalix committed:
[IMP] Module 'auditlog' - Add a test regarding a resource deleted through a (2, ID) tuple on a write operation

sebalix committed:
[FIX] Module 'auditlog' - When updating a resource, filter IDs on 2many fields to prevent a 'name_get()' call on potential deleted resources

sebalix committed:
[FIX] Module 'auditlog' - When updating a resource, filter IDs on 2many fields to prevent a 'name_get()' call on potential deleted resources

dreispt committed:
Add better explanation

gurneyalex committed:
fix pep8 W503

hbrunn committed:
[ADD] migration script

hbrunn committed:
[ADD] migration script

yvaucher committed:
Merge pull request #130 from gurneyalex/8.0-fix_W503

fix pep8 W503

sebalix committed:
Merge pull request #6 from hbrunn/8.0-auditlog

[ADD] migration script

pedrobaeza committed:
Merge pull request #102 from dreispt/add-shell

Backport v9 shell CLI command

pedrobaeza committed:
[IMP] shell: README

Commit:
Merge pull request #119 from osiell/8.0

[ADD] Module 'auditlog' - A substitute to the deprecated 'audittrail' module for Odoo 8

Commit:
add new module for password settings

Commit:
add new module for password settings

dhp-denero committed:
remove icon

dhp-denero committed:
remove icon

legalsylvain commented on auth_password_settings/data/auth_password_settings_data.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
```
It could be great to set an noupdate option here. Otherwise, the changes of the settings will be erased at each update.

dhp-denero replied:
OK I am doing

legalsylvain commented on auth_password_settings/models/res_users.py:
```diff
@@ -0,0 +1,70 @@
+
+    def signup(self, cr, uid, values, token=None, context=None):
+        import pdb
+        pdb.set_trace()
```
hum ... ;-)

dhp-denero replied:
Sorry It was by mistake. I fix it now

pedrobaeza commented on auth_password_settings/models/__init__.py:
```diff
@@ -0,0 +1,22 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import res_config
```
Use relative import:

```
from . import res_config
```

pedrobaeza commented on auth_password_settings/__init__.py:
```diff
@@ -0,0 +1,21 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import models
```
from . import models

pedrobaeza commented on auth_password_settings/__openerp__.py:
```diff
@@ -0,0 +1,57 @@
+##############################################################################
+#    
+#    Copyright (C) 2011 - TODAY Denero Team. (<http://www.deneroteam.com>)
+#    All Rights Reserved
```
This sentence is deprecated. Please remove it.

dhp-denero replied:
Which line ?

pedrobaeza replied:
"All rights reserved"

pedrobaeza commented on auth_password_settings/__openerp__.py:
```diff
@@ -0,0 +1,57 @@
+    'author': 'Dhaval Patel<dhaval@deneroteam.com>',
+    'website': 'http://www.deneroteam.com',
+    'description': """
+A setting module to manage user password settings
```
Please extract this to a README.rst file and use OCA template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

pedrobaeza commented on auth_password_settings/changelog:
```diff
@@ -0,0 +1,6 @@
+2015-02-24: 1.0
```
This can be part of the README file.

dhp-denero replied:
Do you mean, I put changelog in README and remove changelog file ?

Please correct me

Thanks,

Dhaval Patel
Team Lead Engineer
Denero Team.
www.deneroteam.com
Skype : deneroteam
Mobile : +91-9327694716

On Thu, Feb 26, 2015 at 3:06 PM, Pedro M. Baeza notifications@github.com
wrote:

> In auth_password_settings/changelog
> https://github.com/OCA/server-tools/pull/134#discussion_r25414288:
> 
> > @@ -0,0 +1,6 @@
> > +2015-02-24: 1.0
> 
> This can be part of the README file.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/server-tools/pull/134/files#r25414288.

pedrobaeza commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,55 @@
+    _inherit = 'base.config.settings'
+
+    _columns = {
+        'auth_password_min_character': fields.integer('Minimum Password Length',
```
Please respect PEP8 80 cols max length.

pedrobaeza commented on auth_password_settings/models/res_users.py:
```diff
@@ -0,0 +1,70 @@
+##############################################################################
+#    
+#    Copyright (C) 2011 - TODAY Denero Team. (<http://www.deneroteam.com>)
+#    All Rights Reserved
```
Remove

pedrobaeza commented:
Some remarks. Anyway, very good addition.

Thanks.

bealdav commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,55 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import osv, fields
```
Seems import path is easier like this snippet in v8

``` python
from openerp import (
    models,
    fields,
    api,
    _,
    exceptions)
```

delete unused element

bealdav commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,55 @@
+from openerp.osv import osv, fields
+from openerp.tools.safe_eval import safe_eval
+
+class base_config_settings(osv.osv_memory):
```
maybe better in python class convention

``` python
class BaseConfigSettings(models.TransientModel):
```

and with v8 import

bealdav commented:
Thanks for this really good contribution

dhp-denero committed:
Update changes as per PR review

dhp-denero committed:
Update changes as per PR review

hbrunn commented:
please have a look at the flake8 errors: https://travis-ci.org/OCA/server-tools/jobs/52255448

Commit:
[IMP] : run cron from tree view

Commit:
[IMP] : run cron from tree view

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

yvaucher committed:
Move module icon in static/description

yvaucher committed:
Move module icon in static/description

pedrobaeza committed:
Merge pull request #137 from yvaucher/8.0-module-icons

Move module icon in static/description

gurneyalex committed:
fix travis (flake8)

gurneyalex committed:
fix travis (flake8)

pedrobaeza committed:
Merge pull request #138 from gurneyalex/8.0-fix_travis

fix travis (flake8)

hbrunn committed:
[FIX] browse and search groups with superuser permissions

Commit:
Merge pull request #113 from hbrunn/8.0-auth_dynamic_groups

[ADD] auth_dynamic_groups

pedrobaeza committed:
Merge pull request #135 from ah-taktik/8.0-run-cron-from-tree

[IMP] : run cron from tree view

rdeheele commented on auth_password_settings/README.rst:
```diff
@@ -0,0 +1,22 @@
+Enforce users to set password more complex
+=========================================
+This module will allow adminstrator to set password determination rules.
+TO do this, User need to go Settings / Configuration / General Configuration
```
s/TO/To

rdeheele commented on auth_password_settings/models/res_users.py:
```diff
@@ -0,0 +1,70 @@
+        password_rules.append(
+            lambda s: len(s) >= config_data.get('auth_password_min_character', 6) or _('Has %s or more characters') % (config_data.get('auth_password_min_character', 6)) )
+        if (config_data.get('auth_password_has_capital_letter', False)) :
+            password_rules.append(lambda s: any(x.isupper() for x in s) or _('Has atleast One Capital letter'))
```
s/atleast/at least

dhp-denero replied:
OK I am going to change it

rdeheele commented:
very helpful contribution, thanks!

dhp-denero committed:
fix: spelling mistake

dhp-denero committed:
fix: spelling mistake

dhp-denero committed:
fix: pep8 competible

dhp-denero committed:
fix: pep8 competible

dhp-denero commented:
Thank you very much for appreciation (Y)

dhp-denero committed:
fix: flake8 competible

dhp-denero committed:
fix: flake8 competible

dhp-denero commented:
Hello,

I dont know how to correct ci/runbot test. Can anybody assist me ?

legalsylvain commented:
@dhp-denero, don't loose time on that.
cf. https://twitter.com/fpodoo/status/575459986584678400
CC @max3903 
Regards.

dhp-denero committed:
fix: correct spell

dhp-denero committed:
fix: correct spell

gurneyalex commented:
@dhp-denero @legalsylvain I'm working on it. Sorry for the inconvenience. In this case, I'm not able to simply create the symlink in the runbot environment that is created in the travis configuration. I'll disable the runbot for server-tools for now until the matter is resolved.

rdeheele commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,108 @@
+        if (config.auth_password_min_character < 5):
+            raise osv.except_osv(
+                _('Error!'),
+                _('Password Lenght should not be less then 5.')
```
s/Lenght/Length

dhp-denero committed:
fix: correct spell

dhp-denero committed:
fix: correct spell

dhp-denero committed:
fix: import statement as in v8

dhp-denero committed:
fix: import statement as in v8

gurneyalex commented on auth_password_settings/__openerp__.py:
```diff
@@ -0,0 +1,50 @@
+    'category': 'base',
+    'sequence': 3,
+    'summary': 'User Password Settings',
+    'author': 'Dhaval Patel<dhaval@deneroteam.com>',
```
please add 'Odoo Community Association (OCA)' to the list of authors so that the modules shows up as an OCA addon on apps.odoo.com.

gurneyalex commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,110 @@
+            help="Use capital letters to determine the Capital letter that\
+            must be used in the password "),
+        'auth_password_has_digit': fields.boolean(
+            'Use degits',
```
s/degits/digits/

gurneyalex commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,110 @@
+        'auth_password_has_digit': fields.boolean(
+            'Use degits',
+            help="Use capital letters to determine the digit(numaric letter)\
+            that must be used in the password "),
```
unclear help, partly copy pasted from the help of the previous field (it mentions 'capital letters')

gurneyalex commented on auth_password_settings/models/res_config.py:
```diff
@@ -0,0 +1,110 @@
+        'auth_password_has_special_letter': fields.boolean(
+            'Use Special Characters',
+            help="Use capital letters to determine the special letter (e.g. #,\
+            $,!,^, &) that must be used in the password"),
```
still a mention of 'capital letters'. I suggest "this field".

dhp-denero committed:
fix: spell mistakes

dhp-denero committed:
fix: spell mistakes

gurneyalex committed:
don't use travis container-based infrastructure

dhp-denero committed:
Merge branch '8.0' of https://github.com/dhp-denero/server-tools into 8.0

dhp-denero committed:
fix: validation does not work well

dhp-denero committed:
FIX: Flask8 Competiblity issue

guewen commented:
@dhp-denero hi, do you know why there is so many commits and why the diff includes other things?
It seems that the commits are the same than 8.0 but with a different SHA, did you run an interactive rebase on the whole history?

Could you rebase on 8.0?

guewen commented:
@dhp-denero you should be able to rebase and avoid conflicts with: `git rebase -s recursive -X theirs origin/8.0`

hbrunn commented:
superseded by #294
